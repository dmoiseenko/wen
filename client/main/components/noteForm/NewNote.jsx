import React from 'react';
import PropTypes from 'prop-types';


export default function NewNote({ addNote }) {
  function handleClick(e) {
    e.preventDefault();
    addNote();
  }

  return (
    <p className="control">
      <button
        type="submit"
        className="button is-medium is-primary"
        onClick={handleClick}
      >
        Add Note
      </button>
    </p>
  );
}

NewNote.propTypes = {
  addNote: PropTypes.func.isRequired,
};
