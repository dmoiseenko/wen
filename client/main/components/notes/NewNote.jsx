import React from 'react';
import PropTypes from 'prop-types';


export default function NewNote({ addNote, noteText }) {
  function handleClick(e) {
    e.preventDefault();
    addNote(noteText);
  }

  return (
    <p className="control">
      <button
        type="submit"
        className="button is-primary"
        onClick={handleClick}
      >
        Add Note
      </button>
    </p>
  );
}

NewNote.propTypes = {
  addNote: PropTypes.func.isRequired,
  noteText: PropTypes.string.isRequired
};
