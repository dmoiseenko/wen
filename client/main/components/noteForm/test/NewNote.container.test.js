import * as container from '../NewNote.container';
import { addNote as addNoteActionCreator } from '../../../redux/modules/newNote.module';


jest.mock('../../../redux/selector/noteForm.selector');
jest.mock('../../../redux/modules/newNote.module');


test('mapDispatchToProps should map dispatch to props correctly', () => {
  expect(container.mapDispatchToProps).toEqual({
    addNoteActionCreator
  });
});

it('propsMapper should map add note mutation', () => {
  const props = {
    mutate: 'mutate',
    addNoteActionCreator: jest.fn()
  };

  const actual = container.propsMapper(props);

  expect(actual).toMatchSnapshot();
});

it('addNote should call addNoteAction', () => {
  const props = {
    mutate: 'mutate',
    addNoteActionCreator: jest.fn()
  };

  const { addNote } = container.propsMapper(props);
  addNote();

  expect(props.addNoteActionCreator).toHaveBeenCalledWith('mutate');
});
