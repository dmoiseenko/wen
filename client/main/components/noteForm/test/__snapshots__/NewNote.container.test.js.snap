// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propsMapper should map add note mutation 1`] = `
Object {
  "addNote": [Function],
}
`;
